### ISSUE TITLE
Range of scores for MS COCO dataset

### ISSUE THREAD

**Issue by LopezGG** [UKPLab/MMT-Retrieval#6]
Hi,
I am getting scores greater than 1  or less than -1 when I tried to replicate your sample experiment  [here](https://github.com/UKPLab/MMT-Retrieval/blob/master/examples/applications/Image_Search.ipynb). I am surprised because based on your paper I was expecting a probability score. I modified the search function to output scores. My output looks like

![image](https://user-images.githubusercontent.com/1724189/145716383-cfcace03-7109-48ea-acb7-5479aef5afc2.png). 
The  code is 

```
model_path = "https://public.ukp.informatik.tu-darmstadt.de/reimers/mmt-retrieval/models/v1/oscar_join_mscoco.zip"
model = MultimodalTransformer(model_name_or_path=model_path)

feature_file = 'test_img_frcnn_feats.pt' # from the MSCOCO dataset downloaded from OSCAR
if not os.path.exists(feature_file):
    zip_save_path = 'mscoco_test_img_frcnn_feats.zip'
    util.http_get('https://public.ukp.informatik.tu-darmstadt.de/reimers/mmt-retrieval/datasets/mscoco_test_img_frcnn_feats.zip', zip_save_path)
    with zipfile.ZipFile(zip_save_path, 'r') as zip:
        zip.extractall(".")
    os.remove(zip_save_path)


model.image_dict.load_oscar_format_image_features(feature_file)
embedding_storage_file = None
image_searcher = ImageTextRetrieval(images=list(model.image_dict.keys()), embedder=model, cross_encoder=model, embedding_batchsize=32, cross_encoder_batchsize=32)
embeddings = image_searcher.create_embeddings() # the first call to image_searcher.search() also calls this but we do it explicitly in this example

def display_imageid(imageid):
    url = f"http://images.cocodataset.org/train2014/COCO_train2014_{int(imageid):012d}.jpg"
    r = requests.get(url)
    if r.status_code == 404:
        url = f"http://images.cocodataset.org/val2014/COCO_val2014_{int(imageid):012d}.jpg"
    display(IPImage(url=url, width=200))

  def search(query, display_k=3, retrieve_k=10):
      hits = image_searcher.search(text_queries=query, topk=retrieve_k)
      image_ids = hits['images'][0][:display_k]
      scores= hits['scores'][0][:display_k]
      print("Query:")
      display(query)
    
      for image_id,score in zip(image_ids,scores):
          print(image_id + ' : ' +str(score))
          display_imageid(image_id)
```

**Comment by gregor-ge** [comment]
Hi,

Those values are correct because the scores are the raw logits and not the probabilities (see https://github.com/UKPLab/MMT-Retrieval/blob/master/mmt_retrieval/retrieval.py#L117 and https://github.com/UKPLab/MMT-Retrieval/blob/master/mmt_retrieval/model/models.py#L52).

Since the raw logit is roughly proportional to the probabilities, we can skip the softmax when we are only interested in re-ranking. We do that for all reported results.
If you want to know the probabilities (because they are easier to understand for humans or some other reason), you could change the search() function in the ImageTextRetrieval to something like 
```
sims = torch.softmax(self.cross_encoder.encode(captions, images, batch_size=self.cross_encoder_batchsize,
                                                 output_value="logits", show_progress_bar=show_progress_bar, convert_to_tensor=True), dim=-1)[:,-1]


```
A note, p("image-text pair matches") = torch.softmax(self.cross_encoder.encode))[:1] and  p("image-text pair does not matche") = torch.softmax(self.cross_encoder.encode))[:0]

Hope that helps and if you have any other questions, please feel free to ask.

**Comment by LopezGG** [comment]
thank you for the very quick clarification :)
